opm/simulators/flow/GenericCpGridVanguard.cpp

```diff
@@ -550,7 +550,9 @@ void GenericCpGridVanguard<ElementMapper,GridView,Scalar>::doCreateGrids_(Eclips
 }
 
 template<class ElementMapper, class GridView, class Scalar>
-void GenericCpGridVanguard<ElementMapper,GridView,Scalar>::addLgrsUpdateLeafView(const LgrCollection& lgrCollection, const int lgrsSize)
+void GenericCpGridVanguard<ElementMapper,GridView,Scalar>::addLgrsUpdateLeafView(const LgrCollection& lgrCollection,
+                                                                                 const int lgrsSize,
+                                                                                 Dune::CpGrid grid)
```

Comment:
Do you really want to copy here?

I guess this will actually work like expected (because we only do a shallow copy) but it is still copying shared_ptr